src/vcf/validator.cpp

```diff
@@ -122,13 +120,52 @@ namespace ebi
     }
 
     bool is_valid_vcf_file(std::istream &input,
-                           ebi::vcf::Parser &validator,
+                           const std::string &sourceName,
+                           ValidationLevel validationLevel,
+                           Ploidy ploidy,
                            std::vector<std::unique_ptr<ebi::vcf::ReportWriter>> &outputs)
     {
+        std::vector<char> line;
+        ebi::util::readline(input, line);
+        ebi::vcf::Version version = extract_version(line);
+        std::unique_ptr<Parser> validator = build_parser(sourceName, validationLevel, version, ploidy);
+        return is_valid_vcf_file(line, input, *validator, outputs);
+    }
+
+    Version extract_version(std::vector<char> line)
+    {
+        std::string common_substring{"##fileformat=VCFv4."};
+        std::vector<char>::iterator lineIt;
+        std::string::const_iterator commonIt;
+        std::tie(commonIt, lineIt) = std::mismatch(common_substring.begin(), common_substring.end(), line.begin());
+        if (commonIt != common_substring.end()) {
+            // the fileformat is wrong, so take any version and let the parser report the error
```

Comment:
An error should be reported as soon as it happens. Letting the application run only increases chances for wrong behavior.